pkg/util/util_test.go

```diff
@@ -110,3 +110,29 @@ func TestGetRequestHost(t *testing.T) {
 	extHost := GetRequestHost(proxyReq)
 	g.Expect(extHost).To(Equal("external.example.com"))
 }
+
+func TestGetRequestProto(t *testing.T) {
+	g := NewWithT(t)
+
+	req := httptest.NewRequest("GET", "https://example.com", nil)
+	proto := GetRequestProto(req)
+	g.Expect(proto).To(Equal("https"))
+
+	proxyReq := httptest.NewRequest("GET", "http://internal.example.com", nil)
+	proxyReq.Header.Add("X-Forwarded-Proto", "http")
```

Comment:
You might want these to differ so that it's obvious the forwarded proto takes precedence here
```suggestion
	proxyReq := httptest.NewRequest("GET", "https://internal.example.com", nil)
	proxyReq.Header.Add("X-Forwarded-Proto", "http")
```